zstd-safe/src/lib.rs

```diff
@@ -1148,6 +1333,23 @@ impl<'a> InBuffer<'a> {
         InBuffer { src, pos: 0 }
     }
 
+    /// Returns the current cursor position.
+    pub fn pos(&self) -> usize {
+        self.pos
+    }
+
+    /// Sets the new cursor position.
+    ///
+    /// # Panics
+    ///
+    /// If `pos > self.src.len()`.
+    pub fn set_pos(&mut self, pos: usize) {
```

Comment:
If OutBuffer::set_pos should be unsafe, I think this needs to be `unsafe` too since it sets a value for calling that method on drop, right?

Reply:
This is for the `InBuffer`, where we never write to (we only read from that). I don't think we set the `OutBuffer::pos` based on that either.
What we do is set the `OutBuffer::pos` based on `OutBufferWrapper::buf`, a `zstd_sys::ZSTD_outBuffer`, which is updated by the C zstd library, at the same time it actually writes to the buffer itself.